src/OpenTelemetryEngine/Logging/FunctionsLoggerBuilder.cs

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Logs;
+using OpenTelemetryEngine.Types;
+
+
+namespace OpenTelemetryEngine.Logging
+{
+    public class FunctionsLoggerBuilder 
+    {
+        internal static FunctionsLogger? _logger;
+
+        public static void SetLogger()
+        {
+            if (_logger is null)
+            {
+                var loggerFactory = LoggerFactory.Create(builder =>
+                {
+                    builder.AddOpenTelemetry(options =>
+                    {
+                        options.AddOtlpExporter();
+                    });
+                });
+
+                _logger = new FunctionsLogger(loggerFactory.CreateLogger("Azure Functions PowerShell"));
+            }
+        }
+        
+        public static FunctionsLogger GetLogger() 
+        {
+            SetLogger();
+
+            if (_logger is null)
```

Comment:
Wondering if this check is valuable. _logger is not passed to this method externally, we've just created a new instance in the SetLogger method, which we fully control. What would possibly lead to this condition? Consider removing this check.

Reply:
Withdrawing this comment, as I believe I misunderstood this code when I made the comment.